<launch>

  <arg name="teleop_param"  default="$(find-pkg-share teleop_manager)/config/teleop.param.yaml"/>
  <arg name="autonomous_control_topic" default="/ackermann_cmd"/>
  <arg name="vehicle_control_topic" default="/jetracer/cmd_drive"/>
  <arg name="trigger_topic" default="/rosbag2_recorder/trigger"/>
  <arg name="memo_topic" default="/rosbag2_recorder/memo"/>

  
  <!-- joynode -->
  <node pkg="joy" 
        exec="joy_node"
        name="joy_node"
        output="screen">
    <param name="deadzone" value="0.05"/>
  </node>

  <!-- joy manager node-->
  <node pkg="teleop_manager"
        exec="teleop_manager_node"
        name="teleop_manager_node"
        output="screen">
        
    <param from="$(var teleop_param)"/>
    <!-- joy -->
    <remap from="/joy" to="/joy"/>
    <!-- AI制御コマンド -->
    <remap from="/ackermann_cmd" to="$(var autonomous_control_topic)"/>
    <!-- RCカーへの入力 -->
    <remap from="/cmd_drive" to="$(var vehicle_control_topic)"/> 
    <!-- rosbag制御トリガ -->
    <remap from="/rosbag2_recorder/trigger" to="$(var trigger_topic)"/>
    <!-- rosbagメモ -->
    <remap from="/rosbag2_recorder/memo" to="$(var memo_topic)"/>
  </node>

</launch>